modules/apps/oauth2-provider/oauth2-provider-rest-spi/src/main/java/com/liferay/oauth2/provider/rest/spi/web/key/provider/WebKeyProvider.java

```diff
@@ -0,0 +1,24 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.oauth2.provider.rest.spi.web.key.provider;
+
+/**
+ * @author Stian Sigvartsen
+ */
+public interface WebKeyProvider {
+
```

Comment:
The idea is nice, but the problem is we will have portal key management system down the line. This abstraction and key configurations will eventually be replaced, or at least reworked, and between now and then, we have to maintain additional complexity.

Also what happens when devs create their own webKeyProvider?

Reply:
Interestingly this is one of the main reasons why I thought an SPI here would be particularly useful. The portal key management system would simply register a `WebKeyProvider` to retrieve the web key from itself. Eating our own dog food.

Reply:
Also we can track all `WebKeyProvider` services to return for the `.well-known/jwks.json` endpoint. It can track all for a `companyId`, or just the ones with `((companyId=*)(!(oauth.token.issuer=*)))` if we only want it to reveal publicly the web key from the configuration.